lib/DxcSupport/FileIOHelper.cpp

```diff
@@ -171,6 +175,38 @@ UINT32 DxcCodePageFromBytes(const char *bytes, size_t byteLen) throw() {
   return codePage;
 }
 
+unsigned GetBomLengthFromCodePage(UINT32 codePage) {
+  switch(codePage) {
+    case CP_UTF8:
+      return 3;   // UTF-8
+    case 12000:
+    case 12001:
+      return 4;   // UTF-32
+    case 1200:
+    case 1201:
```

Comment:
They are listed here:
https://docs.microsoft.com/en-us/windows/win32/intl/code-page-identifiers
but do not seem to be defined in any available includes that I can find.

This code already had hard-coded values in BOM detection that I copied, but I could have crafted some local defines for use here.

Reply:
I pushed an update to add defines for these locally.

It also explicitly catches UTF cases we can detect with BOM, but can't translate for use, and returns a better HRESULT (before, it would have returned a translated ERROR_BAD_ARGUMENTS, IIRC).  Unfortunately, there would be more plumbing required to preserve this detail for includes through dxcfilesystem/MSFileSystem/HeaderSearch, but that can be tackled later if deemed useful.